DSCResources/MSFT_xWaitForADDomain/MSFT_xWaitForADDomain.psm1

```diff
@@ -80,17 +84,38 @@ function Test-TargetResource
         [UInt32]$RetryCount = 5
     )
 
-    Write-Verbose -Message "Checking for domain $DomainName ..."
-    try
+    $domain = Get-Domain @PSBoundParameters
+    if ($domain.name)
     {
-        $domain = Get-ADDomain -Identity $DomainName -Credential $DomainUserCredential
         Write-Verbose -Message "Found domain $DomainName"
         $true
     }
-    Catch
+    else
     {
         Write-Verbose -Message "Domain $DomainName not found"
         $false
     }
 }
 
+function Get-Domain
+{
+    [OutputType([System.Boolean])]
+    param
+    (
+        [Parameter(Mandatory)]
+        [String]$DomainName,
+
+        [Parameter(Mandatory)]
+        [PSCredential]$DomainUserCredential,
+
+        [UInt64]$RetryIntervalSec = 10,
+
+        [UInt32]$RetryCount = 5
+    )
+    Write-Verbose -Message "Checking for domain $DomainName ..."
+    New-Object DirectoryServices.DirectoryEntry(
```

Comment:
No. Even when setting $ErrorActionPreference = 'Stop', it doesn't throw a terminating error. If it errors out, it will just return $null.